src/pulse3D/excel_writer.py

```diff
@@ -456,10 +519,31 @@ def _write_xlsx(
         for i_col_idx, i_col_width in ((0, 25), (1, 40), (2, 25)):
             ws.set_column(i_col_idx, i_col_idx, i_col_width)
 
+        # stimulation protocols
+        if include_stim_protocols:
```

Comment:
Based on how you set `stim_protocols_df` above, I think you could just check `if stim_protocols_df:` here. Also you'll need to handle the case where `stim_protocols_df = pd.DataFrame({"No stimulation protocols have been used for this recording"})` to avoid the error I mentioned that happens on this line `subprotocols = protocol_data["Subprotocols"]` when no protocols are set